packages/domain/lib/use_case/request_details_use_case.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:domain/base/mappers/mapper_people_model.dart';
+import 'package:domain/entity/cast_response/cast_response.dart';
+import 'package:domain/entity/cast_response/tmdb_people_response.dart';
+import 'package:domain/model/response_model_people.dart';
+import 'package:domain/repository/network_tmdb_repository.dart';
+import 'package:domain/repository/network_trakt_repository.dart';
+import 'package:domain/use_case/sample_use_case/use_case_in_out.dart';
+
+const _maxLengthPeople = 4;
+
+class RequestDetailsUseCase
+    extends UseCaseInOut<int, Future<List<ResponseModelPeople>>> {
+  final NetworkTraktRepository _networkRepositoryTrakt;
+  final NetworkTMDBRepository _networkRepositoryTMDB;
+  final MapperPeopleModel _mapperPeopleModel;
+
+  RequestDetailsUseCase(
+    this._networkRepositoryTrakt,
+    this._networkRepositoryTMDB,
+    this._mapperPeopleModel,
+  );
+
+  @override
+  Future<List<ResponseModelPeople>> call(int params) async {
+    final responseCast = await _networkRepositoryTrakt.requestMoviePeople(
+      id: params,
+    );
+    final cast = responseCast.cast ?? List.empty();
+    final peopleLength =
+        cast.length >= _maxLengthPeople ? _maxLengthPeople : cast.length;
+    final sortedCast = cast.take(peopleLength).toList();
+    final List<TMDBPeopleResponse> responseTMDBPerson =
+        await requestTMDBImage(sortedCast);
+    return _mapperPeopleModel(
+      cast,
+      responseTMDBPerson,
+    );
+  }
+
+  Future<List<TMDBPeopleResponse>> requestTMDBImage(
+    List<Cast> cast,
+  ) async {
+    return await Future.wait(
+      cast.map(
+        (e) async => await _networkRepositoryTMDB.requestPersonTMDB(
+          id: e.person?.ids?.tmdb ?? 0,
+        ),
+      ),
+    );
```

Comment:
1) (optional) return -> =>
2) need to think over the mapping logic

now u have the following flow

1) receive the list
2) for each item u call the api to get the image ( O(n) )
3) add response from image to list with images responses
4) map 2 lists ( O(n^2) - because while mapping u iterate the list on each element)

but at the same time, if u will map the image response with cast right after u received image response, your complexity will be O(n)